Update Haskell package set to Stackage Nightly 2021-04-15 (plus other fixes)

peti opened the pull request:
This PR is test-built by Hydra at https://hydra.nixos.org/jobset/nixpkgs/haskell-updates. I'll fix up the remaining errors and merge it on **Friday, 2021-04-23 20:00 UTC+2**. Everyone can participate in that process live on Twitch at https://www.twitch.tv/peti343. In addition to the chat features offered by Twitch, there is also a voice conference at https://discord.gg/YTEa3XR viewers can use to chat with me and with each other.

peti committed:
Stackage Nightly 2021-04-15

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/604db45cba2a8dcd9a7b3ac991e1bfef8c2eb116.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/613590b5c0f510be99f7b4fcadedab5f2d77f2e1.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/917f9a29ffc883b19a0400f979559c0eb0f2ed29.

LuisChDev committed:
haskellPackages.{H, inline-r}: not broken

These packages were marked as broken but build fine and pass their
respective tests.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/06569db4d571971f79a20750304d929d18725bde.

Gabriella439 committed:
haskellPackages.parameterized: Re-enable without tests

The package builds without the test suite

The upstream issue is: https://github.com/louispan/parameterized/issues/2

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/56696706bb3174f2f2a6b135a2129ffdbc76599c.

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/c47aee4f706eae86dae6efc3d823c8094460e088.

magthe committed:
haskellPackages.servant-docs: unbreak

cdepillabout committed:
haskellPackages.cborg: remove jailbreak

cdepillabout committed:
spago: 0.20.0 -> 0.20.1

hmenke committed:
wstunnel: deduplicate by using haskellPackages

wstunnel is packaged twice in nixpkgs, once standalone and once in
haskellPackages. The standalone version is outdated while the version in
haskellPackages is updated automatically from Hackage, so we are
removing the standalone Nix expression but still keep the static
executable.

cdepillabout committed:
Merge pull request #120061 from cdepillabout/update-spago-0.20.1

spago: 0.20.0 -> 0.20.1

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/24b98b95afa0abe0088c3784a15a9b532fc8d733.

sternenseemann committed:
haskellPackages.essence-of-live-coding-warp: fix eval

Use latest http-client version (0.7.8).

maralorn commented:
@peti Just fyi, after it slipped through last week I have fixed the build failure of reflex-dom on master expecting that you would merge master into haskell-updates some time this week. I don‘t  dare cherry-picking the commit to haskell-updates now, because I fear the merge conflicts. I thought I‘d mention it, so you know and don‘t mark that package broken.

peti committed:
Merge origin/master into haskell-updates.

peti committed:
hackage2nix: update list of broken packages to fix evaluation errors

peti committed:
hackage-packages.nix: automatic Haskell package set update

This update was generated by hackage2nix v2.17.0-8-ge18310f from Hackage revision
https://github.com/commercialhaskell/all-cabal-hashes/commit/2fedb31fe1abdbfb2f224486b4a8b689af85024d.

peti committed:
ShellCheck: apply patch to fix Haddock errors

https://github.com/koalaman/shellcheck/issues/2216

peti commented:
We'll merge https://github.com/NixOS/nixpkgs/pull/120446 next week.
